src/Build/Logging/BinaryLogger/ProjectImportsCollector.cs

```diff
@@ -53,35 +34,34 @@ public byte[] GetAllBytes()
         // this will form a chain of file write tasks, running sequentially on a background thread
         private Task _currentTask = Task.CompletedTask;
 
-        public ProjectImportsCollector(string logFilePath, bool createFile, string sourcesArchiveExtension = ".ProjectImports.zip")
+        public ProjectImportsCollector(string logFilePath, string sourcesArchiveExtension = ".ProjectImports.zip")
         {
+            // Archive file is stored in MSBuild temp folder with same name as binlog, but with different extension
+            ArchiveFilePath = Path.Combine(
+                FileUtilities.TempFileDirectory,
```

Comment:
We have a specific method to find the folder to store files just like this that are used to relieve memory pressure (build requests and configurations, today).

```suggestion
                FileUtilities.GetCacheDirectory(),
```

(if you peek behind the scenes that's like one subdirectory deep in TEMP, and disambiguates by PID which is important in case someone builds twice concurrently.)